dev/io.openliberty.http.monitor/src/io/openliberty/http/monitor/HttpStatsMonitor.java

```diff
@@ -0,0 +1,269 @@
+		}
+
+	}
+
+	public static HttpStatsMonitor getInstance() {
+		if (instance != null) {
+			return instance;
+		} else {
+			if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()){
+				Tr.debug(tc, String.format("No instance of %s found", HttpStatsMonitor.class.getName()));
+			}
+
+		}
+		return null;
+	}
+
+	@PublishedMetric
+	public MeterCollection<HttpStats> HttpConnByRoute = new MeterCollection<HttpStats>("HttpMetrics", this);
+
+	@ProbeAtReturn
+	@ProbeSite(clazz = "com.ibm.ws.http.dispatcher.internal.channel.HttpDispatcherLink", method = "sendResponse", args = "com.ibm.wsspi.http.channel.values.StatusCodes,java.lang.String,java.lang.Exception,boolean")
+	public void atSendResponseReturn(@This Object probedHttpDispatcherLinkObj) {
+
+		
+		long elapsedNanos = System.nanoTime() - tl_startNanos.get();
+		HttpStatAttributes retrievedHttpStatAttr = tl_httpStats.get();
+
+		if (retrievedHttpStatAttr == null) {
+			if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()){
+				Tr.debug(tc, "Unable to retrieve HttpStatAttributes. Unable t time.");
+			}
+			return;
+		}
+
+		updateHttpStatDuration(retrievedHttpStatAttr, Duration.ofNanos(elapsedNanos));
+
+	}
+	
+	/**
+	 * Resolve Network Protocol Info  - move to common utility package
+	 * @param protocolInfo
+	 * @param httpStat
+	 */
+	private void resolveNetwortProtocolInfo(String protocolInfo, HttpStatAttributes httpStat) {
+		String[] networkInfo = protocolInfo.trim().split("/");
+		
+	
+		String networkProtocolName = null;
+		String networkVersion = "";
+		if (networkInfo.length == 1) {
+			networkProtocolName = networkInfo[0].toLowerCase();
+		} else if (networkInfo.length == 2) {
+			networkProtocolName = networkInfo[0];
+			networkVersion = networkInfo[1];
+		} else {
+			if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+				Tr.debug(tc, String.format("More values than expected when parsing protocol information: [%s]", protocolInfo) );
+			}
+		}
+
+		httpStat.setNetworkProtocolName(networkProtocolName);
+		httpStat.setNetworkProtocolVersion(networkVersion);
+	}
+	
+	@ProbeAtEntry
+	@ProbeSite(clazz = "com.ibm.ws.http.dispatcher.internal.channel.HttpDispatcherLink", method = "sendResponse", args = "com.ibm.wsspi.http.channel.values.StatusCodes,java.lang.String,java.lang.Exception,boolean")
+	public void atSendResponse(@This Object probedHttpDispatcherLinkObj, @Args Object[] myargs) {
+		
+		tl_httpStats.set(null);; //reset just in case
+		
+		tl_startNanos.set(System.nanoTime());
+		HttpStatAttributes httpStatAttributes = new HttpStatAttributes();
+
+		/*
+		 *  Get Status Code (and Exception if it exists)
+		 */
+		if (myargs.length > 0) {
+			if (myargs[0] != null && myargs[0] instanceof StatusCodes) {
+				StatusCodes statCode = (StatusCodes) myargs[0];
+				httpStatAttributes.setResponseStatus(statCode.getIntCode());
+			}
+
+			if (myargs[2] != null && myargs[2] instanceof Exception) {
```

Comment:
Will there be a chance of ArrayIndexOutOfBound since it checks `myargs.length > 0` only?

Reply:
The args, are actually the parameters of the probed method. So it will always have the 3 args.